.github/workflows/go.yml

```diff
@@ -5,15 +5,13 @@ jobs:
     name: Build
     runs-on: ubuntu-latest
     steps:
-      - name: Set up Go 1.13
-        uses: actions/setup-go@v1
-        with:
-          go-version: 1.13
-        id: go
+      - name: Checkout Repository
+        uses: actions/checkout@v4
 
-      - uses: actions/checkout@v2
+      - name: Setup Golang Environment
+        uses: actions/setup-go@v5
+        with:
+          go-version: stable
```

Comment:
This repo doesn't get a huge amount of development activity. So I can imagine the code using older language features and not getting updated regularly. It would probably be better to use a fixed version here and only upgrade when there is a benefit. Otherwise I'll be fighting fires every now and then.

Reply:
I set it to `1.23` and also merged the build and test workflows into a single CI one. I think it makes it easier to maintain, let me know.

Reply:
I'm not sure I understand the benefit of merging the workflows into one large file. To me it makes things harder to read. I would prefer to keep them as separate workflows.

Reply:
For example you can see that everything is on the same version at a glance. Or you can skip running the integration tests if the build or unit tests fail, as there's clearly something wrong and there's no point in running the integration tests.

I can revert them later today if you don't like it.

Reply:
I'd prefer to revert if that's ok.

Reply:
okay I've reverted the changes.